FortranCode/meso_approx.f90

```diff
@@ -115,154 +114,171 @@ subroutine populate_allstates(this)
     
         implicit none
         class (approximation) :: this
-        integer i, j, dec, count
+        integer i, j, dec, upper_range
```

Comment:
Can we change this variable's name to `nstates` (or `allstatescount`) throughout the code, to make it apparent that this is the number of possible states that the cluster may be found in? (Better to use a "telling" name for this variable!)... 

This could also be added as a field in the datastructure (`this%allstatescount`)...

Reply:
Perfect, I've renamed `upper_range` to `allstatescount`.  I don't think it's a good idea to make `allstatescount` as a member of the `approximation` type in the current code, as it will require updating `allstatescount` every time you define/modify `nstates`. This will simply complicate setting up the object of `approximation`. 
It will be much better to implement a sort of getters and setters in `meso_approx` and all derived types and avoid using any public member of the types. Then, `nstates` can be defined using something like `set_nstates()`, and `allstatescount ` can be updated within that function.